transport/x/redis/redis5.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package redis
+
+import (
+	"errors"
+	"time"
+
+	redis5 "gopkg.in/redis.v5"
+)
+
+type redis5Client struct {
+	addr   string
+	client *redis5.Client
+}
+
+// NewRedis5Client creates a new Client implementation using gopkg.in/redis.v5
+func NewRedis5Client(addr string) Client {
+	return &redis5Client{addr: addr}
+}
+
+func (c *redis5Client) Start() error {
+	c.client = redis5.NewClient(
+		&redis5.Options{Addr: c.addr},
+	)
+
+	return c.client.Ping().Err()
+}
+
+func (c *redis5Client) Stop() error {
+	return c.client.Close()
+}
+
+func (c *redis5Client) LPush(queueKey string, item []byte) error {
+	cmd := c.client.LPush(queueKey, item)
```

Comment:
Yeah, we should check if the client has been started. An atomic bool might do the trick.